### PR TITLE
transaction cleanup cli

### PR DESCRIPTION
**Pull request by clockfort**

### CODE REVIEW

File: atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/AtlasCleanTransactionRange.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+
+        ClosableIterator<RowResult<Value>> range = kvs.getRange(
+                TransactionConstants.TRANSACTION_TABLE,
+                RangeRequest.builder()
+                        .startRowInclusive(TransactionConstants.getValueForTimestamp(startTimestamp))
+                        .build(),
+                Long.MAX_VALUE); // currently we store this list of transactions (and indeed, anything stored with putUnlessExists) at TS=0
+
+        Multimap<Cell, Long> toDelete = HashMultimap.create();
```
Comment (schlosna) [inline atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/AtlasCleanTransactionRange.java]: Can we fit all the deletes in memory?

Comment (clockfort) [inline atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/AtlasCleanTransactionRange.java]: Yeah, I thought about this already. I wrote this CLI initially doing the deletes inside the range scanning section (and keeping nothing in memory at all, a la "get a key, delete a key"), but that is a little awkward for other reasons (the deletes would't be batched / everything would be much slower, and it's not easy to make the same user prompting / reporting)

I expect them to fit in memory, this should just represent a day of transactions that we're talking about here, at the maximum.

**Comment by rjullman** [comment]
Also, we should decide on a naming convention (e.g. I just renamed the AtlasTimestampCommand to TimestampCommand -- I somewhat like not including Atlas).  This is not important, but just a thought for the future.

**Comment by tpetracca** [comment]
Agreed on the naming convention not including Atlas.  Do we need to fast forward the timestamp at the end of this?  It's necessary in PG, but I think it's only necessary because I don't have a dedicated timestamp server there right?  What level of guarantee do I have between the stored upper limit in the TimestampBoundStore vs. the timestamps seen in the transactions table at the time of a restore?  I have a feeling (but am not 100%) that we can just assume the stored upper limit is higher that any value in transactions table.  Also @tmgordeeva

**Comment by rjullman** [comment]
this is basically read to go, but let's wait to merge the pending CLI prs until the testing setup merges and we add tests

**Comment by tpetracca** [comment]
@rjullman given what you said this morning about pushing the CLI stuff through can we get this merged assuming the checks pass?

**Comment by rjullman** [comment]
After a rebase (things have changed with clis) and adding a test (should be doable now) I'm good with this.

**Pull request merged**